### PR TITLE
#3028 Add connector, pipeline API tests

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3028)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/api/connectorApi.test.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+      name: connectorName,
+    },
+  };
+  return connector;
+};
+
+describe('Connector API', () => {
+  beforeEach(() => deleteAllServices());
+
+  it.only('createConnector', async () => {
```
Comment (wu87988622) [inline ohara-manager/client/cypress/api/connectorApi.test.js]: please remove only thanks~~

Comment (saivirtue) [reply ohara-manager/client/cypress/api/connectorApi.test.js]: sorry :(

Comment (konekoya) [reply ohara-manager/client/cypress/api/connectorApi.test.js]: Hmm, isn't this caught by our linter? We have a rule specifically disabled this...

File: ohara-manager/client/cypress/api/pipelineApi.test.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+describe('Pipeline API', () => {
+  beforeEach(() => deleteAllServices());
+
+  it('createPipeline', async () => {
+    const pipeline = await generatePipeline();
+    const result = await pipelineApi.create(pipeline);
+    const { flows, name, group, objects, lastModified, tags } = result;
+
+    expect(flows).to.be.an('array');
+    expect(flows.sort()).to.be.deep.eq(pipeline.flows.sort());
```
Comment (konekoya) [inline ohara-manager/client/cypress/api/pipelineApi.test.js]: Just wondering if the sorting here is necessary?

File: ohara-manager/client/src/api/utils/waitUtils.js [hunk 3]
```diff
@@ -31,6 +31,16 @@ export const waitForStop = res => {
   return state === undefined;
 };
 
+export const waitForConnectRunning = res => {
```
Comment (konekoya) [inline ohara-manager/client/src/api/utils/waitUtils.js]: Is this wait for a connector until it's running?

File: ohara-manager/client/src/api/pipelineApi.js [hunk 4]
```diff
@@ -39,29 +35,24 @@ export const update = async params => {
...
 };
 
 export const getAll = async (params = {}) => {
   const res = await axiosInstance.get(url + URL.toQueryParameters(params));
-  return lodashGet(res, 'data.result', []);
+  return res ? responseUtil(res, pipeline) : [];
 };
 
-export const refresh = async (params = {}) => {
+export const refresh = async params => {
```
Comment (konekoya) [inline ohara-manager/client/src/api/pipelineApi.js]: What does this method do? Is this a "update" method?

File: ohara-manager/client/src/api/utils/definitionsUtils.js [hunk 5]
```diff
@@ -65,12 +65,21 @@ const getTypeWithValueType = key => {
...
-  const connectorDefinition = params.allConnectorDefinitions
-    .filter(param => param.state === 'RUNNING')
-    .map(param => param.connectorDefinitions)
-    .flatMap(param => param)
-    .filter(param => param.className === params.className);
-  return getCluster(connectorDefinition[0]);
+  let connectorDefinition = {};
+  if (params.state === 'RUNNING') {
+    connectorDefinition = params.connectorDefinitions
+      .reduce((acc, cur) => acc.concat([cur]), [])
```
Comment (konekoya) [inline ohara-manager/client/src/api/utils/definitionsUtils.js]: FYI: `acc.concat(cur)` should work just fine :)

File: ohara-manager/client/cypress/api/pipelineApi.test.js (hunk 2 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/api/pipelineApi.test.js]: The "array" object in backend could not guarantee the order of elements. Here I just want to make sure we won't expect failed of wrong order data :)

File: ohara-manager/client/src/api/utils/waitUtils.js (hunk 3 above)
Comment (saivirtue) [reply ohara-manager/client/src/api/utils/waitUtils.js]: Yes~

File: ohara-manager/client/src/api/pipelineApi.js (hunk 4 above)
Comment (saivirtue) [reply ohara-manager/client/src/api/pipelineApi.js]: this function maps to `/v0/api/pipeline/refresh` API, excerpts from [doc](https://ohara.readthedocs.io/en/latest/rest/rest-pipelines.html#refresh-a-pipeline):
> Requires Ohara Configurator to cleanup nonexistent objects of pipeline.

File: ohara-manager/client/src/api/utils/definitionsUtils.js (hunk 5 above)
Comment (saivirtue) [reply ohara-manager/client/src/api/utils/definitionsUtils.js]: OK, modify right now

File: ohara-manager/client/cypress/api/pipelineApi.test.js (hunk 2 above)
Comment (konekoya) [reply ohara-manager/client/cypress/api/pipelineApi.test.js]: Thanks for the detail explanation!

**Pull request merged**